Feeder audit 2026-05-29: wsdot

## Feeder audit summary for `wsdot`

This sweep covers the objective repo-state parts of the feeder-release-checklist: required file presence, catalog + ghpages source entries, source-local pytest with generated producer src paths injected into PYTHONPATH, Docker E2E matrix coverage, ARM static audit, Fabric static audit, and ARM generator dry-run no-diff. It does not rerun live Azure/Fabric deploy bullets, specialist-agent reviews, or upstream API re-audits.

## Findings

- **ERROR** `pytest-failed` — 1 finding(s)
  - Source-local pytest failed. (feeders/wsdot/tests)
- **WARNING** `arm-stub-description` — 1 finding(s)
  - azure-template-with-eventhub.json: parameter 'eventHubSku' has stub or missing metadata.description (feeders/wsdot/azure-template-with-eventhub.json)

## Check evidence

### `arm_audit` — rc=0, 1.26s

```text
ARM template static audit
  RepoRoot:        C:\git\real-time-sources
  Feeders scanned: 1
  Templates:       7
  Blockers:        0
  Warnings:        1

[wsdot] blockers=0 warnings=3
  azure-template-with-eventhub.json
    [WARNING] stub-description: parameter 'eventHubSku' has stub or missing metadata.description
PASS (with 1 warning(s)).
Report written to C:\Users\clemensv\.copilot\session-state\cb5ea344-87f5-406e-a13a-a66811763074\files\fleet-release-audit\arm\wsdot.json
```

### `arm_generator` — rc=0, 2.95s

```text
C:\git\real-time-sources\tools\.gen-out\feeders\wsdot\azure-template-mqtt.json
C:\git\real-time-sources\tools\.gen-out\feeders\wsdot\azure-template-with-eventgrid-mqtt.json
C:\git\real-time-sources\tools\.gen-out\feeders\wsdot\azure-template-with-servicebus.json
C:\git\real-time-sources\tools\.gen-out\feeders\wsdot\infra\azure-template-amqp.json
C:\git\real-time-sources\tools\.gen-out\feeders\wsdot\infra\azure-template-with-servicebus.json
Description sources: total=177, container=31, readme=0, argparse=28, fallback=118
Generated 5 ARM template(s).
```

### `fabric_audit` — rc=0, 1.0s

```text
[32;1mSource[0m[32;1m OptIn[0m[32;1m Notebook[0m[32;1m  Hook[0m[32;1m PASS[0m[32;1m WARN[0m[32;1m BLOCK[0m
[32;1m------[0m [32;1m-----[0m [32;1m--------[0m [32;1m ----[0m [32;1m----[0m [32;1m----[0m [32;1m-----[0m
wsdot   True     True False   10    0     0

Sources: 1 total, 0 blockers, 0 warnings
JSON report: C:\Users\clemensv\.copilot\session-state\cb5ea344-87f5-406e-a13a-a66811763074\files\fleet-release-audit\fabric\wsdot.json
```

### `pytest` — rc=1, 3.74s

```text
.............................F
================================== FAILURES ===================================
______________ TestDataClassSerialization.test_reading_roundtrip ______________

self = <tests.test_wsdot.TestDataClassSerialization object at 0x000002391EBE7950>

    def test_reading_roundtrip(self):
        reading = WSDOTApi.parse_reading(SAMPLE_FLOW_DATA)
        json_str = reading.to_json()
        restored = reading.from_json(json_str)
        assert restored.flow_data_id == reading.flow_data_id
>       assert restored.flow_reading == reading.flow_reading
E       AssertionError: assert <FlowReadingenum.WideOpen: 'WideOpen'> == 'WideOpen'
E        +  where <FlowReadingenum.WideOpen: 'WideOpen'> = TrafficFlowReading(flow_data_id='1234', station_name='I-5 at Seneca', region=<RegionEnum.Northwest: 'Northwest'>, flow_reading=<FlowReadingenum.WideOpen: 'WideOpen'>, reading_time='2021-04-01T00:00:00+00:00').flow_reading
E        +  and   'WideOpen' = TrafficFlowReading(flow_data_id='1234', station_name='I-5 at Seneca', region='Northwest', flow_reading='WideOpen', reading_time='2021-04-01T00:00:00+00:00').flow_reading

tests\test_wsdot.py:298: AssertionError
=========================== short test summary info ===========================
FAILED tests/test_wsdot.py::TestDataClassSerialization::test_reading_roundtrip
!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!
1 failed, 29 passed in 0.78s
```

## Next step

Fix the findings above, then rerun the sequential feeder-release-checklist audit for this feeder and close this issue with the new evidence.

--- a/feeders/wsdot/tests/test_wsdot.py
+++ b/feeders/wsdot/tests/test_wsdot.py
@@ -295,7 +295,7 @@ class TestDataClassSerialization:
         json_str = reading.to_json()
         restored = reading.from_json(json_str)
         assert restored.flow_data_id == reading.flow_data_id
-        assert restored.flow_reading == reading.flow_reading
+        assert restored.flow_reading.value == reading.flow_reading
 
 
 # ---------------------------------------------------------------------------

